### PR TITLE
Rename and move `site` workspace

### PR DESCRIPTION
**Pull request by kachkaev**
The goal of this PR is to improve alignment with common monorepo standards. We will eventually apply the same structure in [HASH repo](https://github.com/hashintel/hash) too.

### TODO

- ~~Merge https://github.com/blockprotocol/blockprotocol/pull/565~~
- ~~Update Vercel config once PR is reviewed, redeploy for testing~~

### Commands to run locally after pulling `main`

```
mv site/playwright-report apps/site/playwright-report
mv site/public/blocks apps/site/public/blocks
mv site/public/schemas apps/site/public/schemas
mv site/blocks-data.json apps/site/blocks-data.json
mv site/site-map.json apps/site/site-map.json
mv site/.env.local apps/site/.env.local
```

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202540836241707

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -1,10 +1,19 @@
+"app: site":
+  - "apps/site/**/*"
+
+"area: apps":
+  - "apps/**/*"
+
```
Comment (kachkaev) [inline .github/labeler.yml:6]: Needs syncing with #649, whichever PR is merged first

**Review by CiaranMn — APPROVED** [review]

File: .github/labeler.yml (hunk 1 above)
Comment (kachkaev) [reply .github/labeler.yml]: done

File: .nycrc.json [hunk 2]
```diff
@@ -2,6 +2,6 @@
   "extends": "@istanbuljs/nyc-config-typescript",
   "all": true,
   "silent": true,
-  "include": "site",
-  "exclude": ["site/.next", "site/public/blocks", "site/tests"]
+  "include": "apps",
+  "exclude": ["apps/site/.next", "apps/site/public/blocks", "apps/site/tests"]
```
Comment (kachkaev) [inline .nycrc.json:6]: @CiaranMn codecov is still empty, but I've run out of ideas. Not sure what else to try. I guess we'll need to fix codecov in another PR, unless it's some kind of a bug that fixes itself. We moved files in this commit, which may somehow affect path resolution.


When generating `coverage/lcov.info` locally, I get expected coverage, e.g.:

```
SF:apps/site/sentry.client.config.ts
FNF:0
FNH:0
DA:6,9
DA:8,9
DA:14,9
LF:3
LH:3
BRDA:11,0,0,9
BRDA:11,0,1,9
BRF:2
BRH:2
```

`DA:6,9` means ‘line 6 covered 9 times’.

Screenshot 2022-10-07 at 00 55 42

When I upload this file as an artifact via codecov action, `DA:6,9` becomes `DA:6,0`:

```
SF:apps/site/sentry.client.config.ts
FNF:0
FNH:0
DA:6,0
DA:8,0
DA:14,0
LF:3
LH:0
BRDA:11,0,0,0
BRDA:11,0,1,0
BRF:2
BRH:0
end_of_record
TN:
```

🤷

**Review by CiaranMn — APPROVED** [review]
Let's merge and see what happens with codecov

File: .nycrc.json (hunk 2 above)
Comment (kachkaev) [reply .nycrc.json:6]: Test coverage was reported on main at ≈40%, which confirms a potential bug in codecov. I’m restoring the original ≈60% in #654

**Pull request merged**